plugins/generator-1.16.5/forge-1.16.5/templates/block.java.ftl

```diff
@@ -110,6 +110,12 @@ public class ${name}Block extends ${JavaModName}Elements.ModElement {
 				<#if data.tintType == "Grass">
 					BiomeColors.getGrassColor(world, pos) : GrassColors.get(0.5D, 1.0D);
 				<#elseif data.tintType == "Foliage">
+					FoliageColors.getDefault() : BiomeColors.getFoliageColor(world, pos);

```

Comment:
The other way around does not work right if a user wants the foliage colour and not the bio color this as all been tested and works

Reply:
"Foliage" has always meant the biome color since it was added, same as the other tint types. This change breaks all the existing blocks with this tint, because now they won't match the biome color anymore. If you want to add a "Default foliage color", you shouldn't replace the existing option

Reply:
As @AlsoSomeoneElse said, alternatively you need to provide a good justification for the switch and prepare a migration path with a workspace converter

Reply:
The best option would be to leave "Foliage" as is, and add something like "Default foliage" for the new option

Reply:
Agree

Reply:
Right when I put the `BiomeColors.getFoliageColor(world, pos) : FoliageColors.getBirch();` the baisc biome color for leaves takes over what isn't how Birch leaves react but if I remove  `FoliageColors.getBirch();` I get a build error, as it's wanting option : option so should I do birch and if fail default `FoliageColors.getBirch() : FoliageColors.getDefault;`

Reply:
Birch, spruce, and default don't need the world/pos check, so the best way to handle those would be to move those before the check. This way you can do something like
```
if tint type is birch
    return birch
elseif tint type is spruce
    return spruce
elseif tint type is default
    return default
else
    use the current implementation
```

Reply:
```JAVA
<#if data.tintType != "Default foliage" || data.tintType != "Foliage birch" || data.tintType != "Foliage spruce">
	return world != null && pos != null ?
<#else>
	return
</#if>
```

Reply:
something more like 
```
<#if data.tintType == "Default foliage">
    return FoliageColors.getDefault();
<#elseif data.tintType == "Foliage birch">
    return birch
<#elseif data.tintType == "Foliage spruce">
    return spruce
<#else>
    return world != null && pos != null ?
    CURRENT IMPLEMENTATION
</#if>
```

Reply:
Or ok ya that looks cleaner